lib/extract-templates.js

```diff
@@ -0,0 +1,133 @@
+import { parseTemplates } from 'ember-template-imports/lib/parse-templates.js';
+
+export const SUPPORTED_EXTENSIONS = ['.js', '.ts', '.gjs', '.gts'];
+const LOCATION_START = Object.freeze({ line: 0, column: 0 });
+/**
+ * @typedef {object} TemplateInfo
+ * @property {number} line
+ * @property {number} column
+ * @property {string} template
+ * @property {boolean} isEmbedded
+ *
+ * @param {string} moduleSource
+ * @param {string} relativePath
+ *
+ * @returns {TemplateInfo[]}
+ */
+export function extractTemplates(moduleSource, relativePath) {
+  // If no relativePath is present, assuming we might have templates.
+  let mightHaveTemplates = relativePath
+    ? SUPPORTED_EXTENSIONS.some((ext) => relativePath.endsWith(ext))
+    : true;
+
+  if (!mightHaveTemplates) {
+    return [makeTemplateInfo(LOCATION_START, moduleSource)];
+  }
+
+  let parsed = parseTemplates(moduleSource, relativePath, 'template');
+
+  // If after parsing we found no templates and we had no relative path,
+  // then we assume we had a hbs file in input.
+  if (parsed.length === 0 && !relativePath) {
+    return [makeTemplateInfo(LOCATION_START, moduleSource)];
+  }
```

Comment:
It seems vaguely odd to have this be a fallback, I think we expect (at least at the moment) for 99% of the things we run against are going to be stand alone `.hbs` files. Can we move this first, and do an extension check _before_ `let mightHaveTemplates`?

Reply:
If `relativePath` is `undefined` it means we're reading a file from `stdin` which means we don't have any information about the file format. I thought the best approach in this case would be to try and parse the file as if it were a script and then if an empty array is returned, we simply return the entire file.

If a `relativePath` is present, we parse the extensions and return on line 24 if the extensions of that file is not in `SUPPORTED_EXTENSIONS`.

Reply:
Gotcha, makes sense now (with the stdin context).

Reply:
I discovered this because ["returns a "fatal" result object if an error occurs during parsing"](https://github.com/ember-template-lint/ember-template-lint/blob/master/test/acceptance/public-api-test.js#L497) was failing. I will add one more test here that explicitly calls out a missing `filePath`.

Reply:
Just added a number of tests to [`extract-template-test.js`](https://github.com/ember-template-lint/ember-template-lint/pull/2483/files#diff-0435102285f4156c51a7c0b981f321d413251423fa542fd1af521078a8cae2b2). To cover all cases.